### ISSUE TITLE
Possible issue with strip_alpha function

### ISSUE THREAD

**Issue by eschoeller** [Cacti/cacti#5090]
Please take a look at #2662. I'm finding that text such as "15 Percent" and "14401 Watts" work just fine. But when I encounter "12609 Volt-Amps", it causes a problem. I suspect it's because of the '-' being interpreted as part of the numeric value and not the alphanumeric text that should be removed. I'm not sure if there's a graceful way to fix this ... but it would be super nice if it could get fixed :) 
Perhaps only accept a '-' character if it's at the beginning of the string, or, directly adjacent to the beginning of the numeric value.

**Comment by TheWitness** [comment]
@eschoeller, are you using spine?

**Comment by TheWitness** [comment]
This is because you have values like `-20 Amperes` and `10.1e-5 joules`.  There what a pain.

**Comment by netniV** [comment]
There should be a simple regex to validate against a number (positive or negative) or where it's in the extended floating point format and extract that.

**Comment by eschoeller** [comment]
Yes I'm using Spine. I hadn't even considered scientific notation. Indeed very much a pain:( perhaps it's too difficult to fix without introducing more issues that we wouldn't think about.

**Comment by netniV** [comment]
https://en.wikipedia.org/wiki/Scientific_notation

There can be various styles of outputting scientific notation.  So I might just stick to the basics. :D

**Comment by netniV** [comment]
I believe this should work:
```
([-+]*)([0-9]*)([.]{1}[0-9]+)
```
If that returns anything other than one match, the line is invalid.
Example: https://regex101.com/r/3B1NgD/2

Match is the full number
Group 1 is the sign
Group 2 is the whole number
Group 3 is the decimal plus number

**Comment by TheWitness** [comment]
The current `strip_alpha()`, walks the string back until it finds a '-' or other character.  So, we could do a `stripos` to check for scientific notation, set a flag.  If it's not, the '-' on the end of the string can safely be removed.  If it's set, then we strip until we find the first numeric on the back end.

**Comment by TheWitness** [comment]
@eschoeller, Hopefully 1.3 comes fast.  Right now, if this get's fixed sooner than later in `develop`, it should be compatible with 1.2.23.
